roslyn-differences.md

```diff
@@ -0,0 +1,41 @@
+# Inconsistencies between Roslyn and the C# standard
+
+As noted in the [README](README.md), although the C# compiler within [Roslyn](https://github.com/dotnet/roslyn) is the closest compiler we have to a reference implementation, there are areas where its behavior is inconsistent with the standard. This document lists those inconsistencies where they are known, organized by standard section.
+
+TODO: There was talk of a document in the Roslyn repo which mentioned this, but I can't find it.
```

Comment:
There are a number of pages in our team onenote that talk about spec deviations. That isn't publicly available, and AFAIK most (if not all) of them are documented inline in the code where they occur (if the deviation is known, of course).

Reply:
I'd definitely be happier linking to a doc in the Roslyn repo. I do think it's useful to have a doc that collects them in one place (for easy checking) rather than expecting users to hunt around.

If there isn't such a public doc at the moment, are folks happy for us to start it here and then request a transfer when we've collected the low-hanging fruit?